### PR TITLE
Politicians teams join table

### PR DESCRIPTION
**Pull request by elreimundo**

### CODE REVIEW

**Comment by MaxDavila** [comment]
You guys might want to use this for your site:
http://teamcolors.arc90.com/

File: app/controllers/index.rb [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 get '/teams/:id' do
   @queried_team = Team.find(params[:id])
+  @access = access_granted?(@queried_team.id)
```
Comment (zspencer) [inline app/controllers/index.rb:20]: What if users could  `current_user.may_access?(@queried_team)`

This would make it super easy to test the rules around who can access what.

File: app/controllers/index.rb [hunk 2]
```diff
@@ -25,6 +26,11 @@
   erb :user
 end
 
+post '/teams/:team_id/politicians/add/:id' do
```
Comment (zspencer) [inline app/controllers/index.rb:29]: This logic seems like it should belong on the team model.

`team.add_politician(politician) if current_user.may_access?(team)`

File: app/controllers/index.rb [hunk 3]
```diff
@@ -37,7 +43,11 @@
 
 post '/politicians/search' do
   politicos = Politician.where('first_name=?',params[:pol_name].capitalize)
-  .map{|p| "#{p.first_name} #{p.last_name}"}
+    .map do |p| 
+      this_element = "<form method='post' action='/teams/#{current_user.id}/politicians/add/#{p.id}'>"
+      this_element << "<input type='submit' class='add' value='+'>"
+      this_element << "#{p.info}</form>"
+    end
```
Comment (zspencer) [inline app/controllers/index.rb:50]: It looks like you're building a collection of html? Why not just render an erb with `layout:  false`? As it stands, you're doing the job of a template inside of your route; which feels funny

File: app/controllers/index.rb [hunk 4]
```diff
@@ -56,7 +66,14 @@ def auth_hash
...
   end
 
+  def access_granted?(team_id)
+    return false unless current_user
+    current_user.teams.each {|t| return true if t.id == team_id}
+    false
+  end
+
   def current_user
+    return nil if User.all.size<1
```
Comment (zspencer) [inline app/controllers/index.rb:76]: Why is this line necessary?

File: app/models/politician_team.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class PoliticianTeam < ActiveRecord::Base
+  self.table_name='politician_teams'
```
Comment (zspencer) [inline app/models/politician_team.rb:2]: Odd that you need to set this. Indicates something is wrong with your schema.

File: spec/controller_spec.rb [hunk 6]
```diff
@@ -35,14 +35,15 @@
...
+# WE CAN MAKE THIS WORK ONCE WE GIVE THE TESTING ENVIRONMENT ACCESS
+# feature 'team page' do
+#   # create_test_team
+#   scenario 'user can search by name' do
+#     visit('/teams/1')
+#     page.fill_in 'pol_name', with: 'John'
+#     page.click_button 'button'
+#     expect(page).to have_content('Sununu')
+#   end
+# end
```
Comment (zspencer) [inline spec/controller_spec.rb:49]: Why no unit tests for all this code?
